test/services/GetConsignmentServiceSpec.scala

```diff
@@ -0,0 +1,110 @@
+
+    when(graphQLClient.getResult(tokenCaptor.capture(), documentCaptor.capture(), variablesCaptor.capture())).thenReturn(Future(graphqlData))
+    Tuple3(documentCaptor, tokenCaptor, variablesCaptor)
+  }
+
+  def mockFailedResponse(graphQLClient: GraphQLClient[gc.Data, gc.Variables]) = {
+    val documentCaptor: ArgumentCaptor[Document] = ArgumentCaptor.forClass(classOf[Document])
+    val tokenCaptor: ArgumentCaptor[BearerAccessToken] = ArgumentCaptor.forClass(classOf[BearerAccessToken])
+    val variablesCaptor: ArgumentCaptor[Option[gc.Variables]] = ArgumentCaptor.forClass(classOf[Option[gc.Variables]])
+
+    when(graphQLClient.getResult(tokenCaptor.capture(), documentCaptor.capture(), variablesCaptor.capture())).thenReturn(Future.failed(HttpError("")))
+    Tuple3(documentCaptor, tokenCaptor, variablesCaptor)
+  }
+
+  "GetConsignmentService GET" should {
+
+    "Return true when given a valid consignment id" in {
+      val graphQLClient = mock[GraphQLClient[gc.Data, gc.Variables]]
+      val graphQLConfig = mock[GraphQLConfiguration]
+      when(graphQLConfig.getClient[gc.Data, gc.Variables]()).thenReturn(graphQLClient)
+
+      val consignmentId = 999
+
+      val consignmentResponse: gc.GetConsignment = new gc.GetConsignment(UUID.randomUUID(), 1L)
+
+      val captors = mockOkResponse(graphQLClient, Some(gc.Data(Some(consignmentResponse))), List())
+
+      val getConsignment = new GetConsignmentService(graphQLConfig).consignmentExists(consignmentId, new BearerAccessToken("someAccessToken"))
+      val actualResults = getConsignment.futureValue
+
+      actualResults should be(true)
+      verifyCaptors(captors, consignmentId)
+    }
+
+    "Return false if consignment with given id does not exist" in {
+      val consignmentId = 999
+      val graphQLClient = mock[GraphQLClient[gc.Data, gc.Variables]]
+
+      val graphQLConfig = mock[GraphQLConfiguration]
+      when(graphQLConfig.getClient[gc.Data, gc.Variables]()).thenReturn(graphQLClient)
+
+      val captors = mockOkResponse(graphQLClient, Option.empty, List(GraphQLClient.GraphqlError("Error", Nil, Nil)))
+
+      val getConsignment = new GetConsignmentService(graphQLConfig).consignmentExists(consignmentId, new BearerAccessToken("someAccessToken"))
+      val actualResults = getConsignment.futureValue
+      actualResults shouldBe false
+      verifyCaptors(captors, consignmentId)
+    }
+
+    "Return an error when the API has an error" in {
+      val consignmentId = 999
+      val graphQLClient = mock[GraphQLClient[gc.Data, gc.Variables]]
+
+      val graphQLConfig = mock[GraphQLConfiguration]
+      when(graphQLConfig.getClient[gc.Data, gc.Variables]()).thenReturn(graphQLClient)
+
+      val captors = mockFailedResponse(graphQLClient)
+
+      val getConsignment = new GetConsignmentService(graphQLConfig).consignmentExists(consignmentId, new BearerAccessToken("someAccessToken"))
+
+      ScalaFutures.whenReady(getConsignment.failed) { results =>
```

Comment:
You can do `getConsignment.failed.futureValue`, and then it uses the same timeouts as `getConsignment.futureValue` in the other tests.

Reply:
Yep, makes sense